src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -211,37 +212,28 @@ private void applyFilterExpression(FilterExpression filter) {
         parentPanelControl.getGUIController().panelFilterExpressionChanged(this);
     }
 
-    private void indicatePanelLoading(RepoOpeningEvent e) {
-        if (isIndicatorApplicable(e.isPrimaryRepo)) {
-            addPanelLoadingIndication();
-        }
-    }
+    protected abstract void startLoadingAnimationIfApplicable(PrimaryRepoOpeningEvent e);
 
-    private void unindicatePanelLoading(RepoOpenedEvent e) {
-        if (isIndicatorApplicable(e.isPrimaryRepo)) {
-            removePanelLoadingIndication();
-        }
-    }
+    protected abstract void stopLoadingAnimationIfApplicable(PrimaryRepoOpenedEvent e);
 
-    private boolean isIndicatorApplicable(boolean isPrimaryRepo) {
-        HashSet<String> allReposInFilterExpr =
-                Qualifier.getMetaQualifierContent(getCurrentFilterExpression(), QualifierType.REPO);
+    protected abstract void startLoadingAnimationIfApplicable(ApplyingFilterEvent e);
 
-        boolean isPrimaryRepoChanged = isPrimaryRepo && allReposInFilterExpr.isEmpty();
+    protected abstract void stopLoadingAnimationIfApplicable(AppliedFilterEvent e);
```

Comment:
Should be better to comment on what its means by `Applicable` here. If more details can be provided based on the concrete method, comments should be added there too.